README.md
max-control
===========

PHP LDAP web interface for Zentyal in Madrid TIC servers


This project is a frontend of Zentyal (now 3.* versions) to manage users, groups, classrooms, and computers in SAMBA4/LDAP

### Install max-control

* Install Zentyal 3
* Add this repo:

```
deb http://max.educa.madrid.org/max-server/max-server-2013 max main
```

* Run this commands:

```
apt-get update
apt-get upgrade
apt-get install max-control
```

* Open browser http://127.0.0.1/control and follow instructions



 * For winexe details: http://sourceforge.net/p/winexe/winexe-waf/ci/master/tree/

debian/changelog
max-control (5.0~max8) max; urgency=medium

  * Fix Computer::DelComputer(), close #24

 -- Mario Izquierdo <hola@marioizquierdo.es>  Tue, 12 Sep 2017 20:08:30 +0200

max-control (5.0~max7) max; urgency=medium

  * fix javascript include files, close #22

 -- Mario Izquierdo <hola@marioizquierdo.es>  Thu, 27 Jul 2017 19:30:34 +0200

max-control (5.0~max6) max; urgency=medium

  * Add password description in import section, rel #19
  * Add A disable calendar for programed actions, close #17

 -- Mario Izquierdo <hola@marioizquierdo.es>  Wed, 26 Jul 2017 23:57:12 +0200

max-control (5.0~max5) max; urgency=medium

  * Fix zentyal-maxcontrol install, close #20
  * Remove ERROR: Conf max-control does not exist! (a2enconf)
  * Fix importer and allow password (optional) in CSV, close #19

 -- Mario Izquierdo <hola@marioizquierdo.es>  Tue, 25 Jul 2017 20:47:07 +0200

max-control (5.0~max4) max; urgency=medium

  * users: Show samaccountname if user is created from Zentyal, close #15
  * INcrease pager size to 50, close #18
  * Users: Revert [profiles] smb.conf and fix delete/create
    profile V2 (<=Win 8) V6 (Win10), close #12, close #13
  * Fix computer save, max-control rights and winexe/reboot issues, close #16

 -- Mario Izquierdo <hola@marioizquierdo.es>  Mon, 24 Jul 2017 23:54:57 +0200

max-control (5.0~max3) max; urgency=medium

  * Update zentyal-maxcontrol
  * Remove [profiles] section from shares.conf, rel #12
  * Add margin-bottom to central container (float footer)
  * Fix: remove profile .V2, close #13
  * Add libpam-cap to depends, close #14
  * Fix creation of groups and add/remove members, close #11
  * bin/pyoverquota: Use multiple REMOTES for calculating quota, close #10

 -- Mario Izquierdo <hola@marioizquierdo.es>  Sun, 23 Jul 2017 16:44:13 +0200

max-control (5.0~max2) max; urgency=medium

  * Remove winexe stuff, use external package

 -- Mario Izquierdo <hola@marioizquierdo.es>  Wed, 19 Jul 2017 00:59:06 +0200

max-control (5.0~max1) max; urgency=medium

  * Build for MAX Server 2017

 -- Mario Izquierdo <hola@marioizquierdo.es>  Wed, 19 Jul 2017 00:34:56 +0200

max-control (3.5.1~max2) max; urgency=low

  * Check if max-control LDAP account is valid
  * Change uid => cn in sort filters
  * Changes to make tftpd-hpa work
  * Set bootstrap as default and remove old templates
  * Add genPXELinux cleanPXELinux to COMPUTER class
  * Call configure-max-control when enabling module

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Thu, 04 Dec 2014 17:36:24 +0100

max-control (3.5.1~max1) max; urgency=low

  * Finish bootstrap
  * Add a welcome wizard to create first user
  * Use more things using samba4 LDAP or EBox API
  * Global clean

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Sun, 05 Oct 2014 15:58:06 +0200

max-control (3.5.0~rc2max1) max; urgency=medium

  * New package with new Bootstrap template
  * Remove winbind as dependence

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Thu, 02 Oct 2014 09:22:36 +0200

max-control (3.5.0~alfa1max2) max; urgency=medium

  * Fix some paths

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Mon, 22 Sep 2014 15:31:41 +0200

max-control (3.5.0~alfa1max1) max; urgency=low

  * Remove zentyal-captiveportal

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Sun, 14 Sep 2014 20:34:43 +0200

max-control (3.5.0~alfa1) max; urgency=low

  * Build in MAX server 2014

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Sun, 14 Sep 2014 20:30:11 +0200

max-control (3.0.0~alfa4) max; urgency=low

  * New alfa, users add/delete/edit/import working

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Thu, 30 May 2013 12:16:47 +0200

max-control (3.0.0~alfa3) max; urgency=low

  * Fix install of zentya-maxcontrol

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Tue, 28 May 2013 16:20:25 +0200

max-control (3.0.0~alfa2) max; urgency=low

  * Prepare alfa2

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Tue, 28 May 2013 12:44:27 +0200

max-control (3.0.0~alfa1) precise; urgency=low

  * New alpha version for Zentyal 3

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Tue, 26 Feb 2013 13:35:57 +0100

max-control (1.1.38) max; urgency=low

  * Only update empty or null MACS

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Wed, 11 Jan 2012 11:45:57 +0100

max-control (1.1.37) max; urgency=low

  * templates/update_equipos.tpl: Use a bit danger message before
    deleteing all MACS

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Tue, 10 Jan 2012 15:50:24 +0100

max-control (1.1.36) max; urgency=low

  * bin/configure-max-control: Don't run if SAMBA module is not enabled

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Wed, 16 Nov 2011 11:19:48 +0100

max-control (1.1.35) max; urgency=low

  * classes/importer.class.php remove quotation before sanitizeOne

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Wed, 26 Oct 2011 02:35:28 +0200

max-control (1.1.34) max; urgency=low

  * Check for spaces in UID and ensure delete dirs with quotation marks

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Wed, 26 Oct 2011 02:24:55 +0200

max-control (1.1.33) max; urgency=low

  * classes/importer.class.php: Only show duplicate error one time
  * templates/import.tpl: Show a message with 20 cut UID
  * Fix LDAP::get_aulas() returning same preffix, close #107

 -- Mario Izquierdo Rodríguez (Thinetic Systems SLNE) <mario.izquierdo@thinetic.es>  Thu, 06 Oct 2011 15:40:35 +0200

max-control (1.1.32) max; urgency=low


conf.inc.php
<?php

// file autogenerated with init.sh on 'mar jul 25 20:31:39 CEST 2017'

// basedn del dominio
define('LDAP_BASEDN', 'DC=madrid,DC=lan');
define('WORKGROUP', 'MADRID');

// autenticacion
define('LDAP_BINDDN', 'CN=max-control,CN=Users,DC=madrid,DC=lan');
define('LDAP_BINDPW', 'ZpLIJ4iIjaCwqxK');

define("LDAP_ADMIN", 'max-control');
// autogenerated password in init.sh
define("LDAP_PASS", 'ZpLIJ4iIjaCwqxK');

// Entidades organizativas
define('LDAP_OU_COMPUTERS', 'CN=Computers,DC=madrid,DC=lan');
define('LDAP_OU_USERS',     'CN=Users,DC=madrid,DC=lan');
define('LDAP_OU_GROUPS',    'CN=Users,DC=madrid,DC=lan');

// dominio
define('LDAP_DOMAIN', 'madrid.lan');

define('LDAP_OU_BUILTINS',      'CN=MAXGroups,DC=madrid,DC=lan');
// Administrators
define('LDAP_OU_ADMINS',        'CN=Administrators,CN=Builtin,DC=madrid,DC=lan');
define('LDAP_OU_DADMINS',       'CN=Domain Admins,CN=Users,DC=madrid,DC=lan');
define('LDAP_OU_DUSERS',        'CN=Domain Users,CN=Users,DC=madrid,DC=lan');


define('TEACHERS', 'Teachers');
define('LDAP_OU_TEACHERS',      'CN=Teachers,CN=MAXGroups,DC=madrid,DC=lan');

define('TICS', 'CoordinadoresTIC');
define('LDAP_OU_TICS',          'CN=CoordinadoresTIC,CN=MAXGroups,DC=madrid,DC=lan');

define('INSTALLATORS', 'Instaladores');
define('LDAP_OU_INSTALLATORS',  'CN=Instaladores,CN=MAXGroups,DC=madrid,DC=lan');


define("HOMES", "/home/MADRID/");

define("LDAP_HOST", "127.0.0.1");
define("LDAP_HOSTNAME", "max-control");
define("LDAP_PORT", 389);

define('CONFIGURED', True);


// aparecerán cajas con información útil para errores.
define("DEBUG", True);

define("VERSION", "5.0~max4");

/*********** a partir de aqui puede que ya no hay aque editar nada ********/

// ruta al comando winexe
define("WINEXE", "/usr/bin/pywinexe");

// ruta al comando max-control
define("MAXCONTROL", "/usr/bin/max-control");

// ruta al comando pywakeonlan
define("PYWAKEONLAN", "/usr/bin/pywakeonlan");

// RUTA a la clase smarty
define("SMARTY_REQUIRE","/usr/share/php/smarty3/Smarty.class.php");

// cache de smart
define("SMARTY_CACHE", "/var/lib/max-control/cache");

// ruta a las plantillas mejor no editar
define("SMARTY_TEMPLATES", "/templates");

define("SMARTY_PLUGINS", '/usr/share/php/smarty3/plugins/');

/*
* enable apache mod_rewrite (mejor no tocar)
*/
define("APACHE_MOD_REWRITE", True);

// puerto usado para conectar por ssh y detectar LINUX
define("LINUX_PORT", 22);
/* esperar 2 segundos para ver si el puerto 22 está abierto*/
define("PROBE_TIMEOUT", 2);

/* timeout para apagar y mostrar un mensaje */
define("POWEROFF_REBOOT_TIMEOUT", 20);

/* directorios TFTP */
define("TFTPBOOT", "/var/lib/tftpboot/");
define("PXELINUXCFG", "/var/lib/tftpboot/pxelinux.cfg/");


// compartir ISOS
define("ISOS_PATH", "/home/samba/shares/isos/");

define("FORK_ACTIONS", True);
//define("FORK_LOGFILE", "/tmp/actions.log");
define("FORK_LOGFILE", "/dev/null");


/* paginador */
define("PAGER_LIMIT", 25);
define("PAGER_MAX_LINKS", 10);

/* quota % to show a warning (needed by pyoverquota) */
define("OVERQUOTA_LIMIT", "80");
define("DEFAULT_QUOTA", "2000");

/* file to read/write programer events */
define("PROGRAMER_INI", "/var/lib/max-control/programer.ini");
define("IMPORTER_DIR", "/var/lib/max-control/importer/");
define("FIRST_RUN", "/var/lib/max-control/first-run.done");

$site["public_modules"]=array();

$site["private_modules_admin"]=array(
        "miperfil" => "Mi perfil",
        "usuarios" => "Usuarios y Grupos",
        "equipos" => "Aulas y Equipos",
        "isos" => "Distribuir ISOS",
        "power" => "Apagado y reinicio",
        "boot" => "Programar arranque",
        );

$site["private_modules_tic"]=array(
        "miperfil" => "Mi perfil",
        "usuarios" => "Usuarios y Grupos",
        "equipos" => "Aulas y Profesores",
        "isos" => "Distribuir ISOS",
        "power" => "Apagado y reinicio",
        "boot" => "Programar arranque",
        );

$site["private_modules_teacher"]=array(
        "miperfil" => "Mi perfil",
        "isos" => "Distribuir ISOS",
        "power" => "Apagado y reinicio",
        );

$site["private_modules_none"]=array(
        "miperfil" => "Mi perfil"
        );

$site["private_modules"]=array();


